File: 0107-binary-tree-level-order-traversal-ii/0107-binary-tree-level-order-traversal-ii.java
class Solution {
    public List<List<Integer>> levelOrderBottom(TreeNode root) {
        List<List<Integer>> result = new ArrayList<>();

        if (root == null) {
            return result;
        }

        Stack<List<Integer>> vals = new Stack<>();
        Queue<TreeNode> que = new ArrayDeque<>();
        que.add(root);

        while (!que.isEmpty()) {
            int n = que.size();
            List<Integer> temp = new ArrayList<>();

            for (int i = 0; i < n; i++) {
                TreeNode now = que.poll();
                temp.add(now.val);

                if (now.left != null) {
                    que.add(now.left);
                }
                if (now.right != null) {
                    que.add(now.right);
                }
            }

            vals.push(temp);
        }

        while (!vals.isEmpty()) {
            result.add(vals.pop());
        }

        return result;
    }
}

File: 0107-binary-tree-level-order-traversal-ii/0107-binary-tree-level-order-traversal-ii.cpp
class Solution {
public:
    vector<vector<int>> levelOrderBottom(TreeNode* root) {
        vector<vector<int>> result;

        if (root == NULL) return result;

        stack<vector<int>> vals;
        queue<TreeNode*> que;
        que.push(root);

        while (!que.empty()) {
            int n = que.size();
            vector<int> temp;

            for (int i = 0; i < n; i++) {
                TreeNode* now = que.front();
                temp.push_back(now->val);
                que.pop();

                if (now->left != NULL) {
                    que.push(now->left);
                }
                if (now->right != NULL) {
                    que.push(now->right);
                }
            }

            vals.push(temp);
        }

        while (!vals.empty()) {
            result.push_back(vals.top());
            vals.pop();
        }

        return result;
    }
};